pulp_ansible/app/models.py

```diff
@@ -266,6 +298,312 @@ class Meta:
+        artifact_signature = await sync_to_async(private_key.sign)(
+            input_digest, ec.ECDSA(Prehashed(hashes.SHA256()))
+        )
+        b64_artifact_signature = base64.b64encode(artifact_signature).decode()
+        rekor_post_entry_payload = {
+            "kind": "hashedrekord",
+            "apiVersion": "0.0.1",
+            "spec": {
+                "signature": {
+                    "content": b64_artifact_signature,
+                    "publicKey": {"content": b64_cert.decode()},
+                },
+                "data": {"hash": {"algorithm": "sha256", "value": input_digest.hex()}},
+            },
+        }
+
+        rekor_post_entries_url = urljoin(self.rekor.url, "entries/")
+        async with aiohttp.ClientSession(
+            headers={"Content-Type": "application/json", "Accept": "application/json"}
+        ) as session:
+            try:
+                async with session.post(
+                    rekor_post_entries_url, json=rekor_post_entry_payload
+                ) as resp:
+                    rekor_response = await resp.json()
+            except aiohttp.web.HTTPError as http_error:
+                raise RekorClientError from http_error
+
+        log_entry = LogEntry._from_response(rekor_response.json())
+        log.info(f"Transparency log entry created with index: {log_entry.log_index}")
+
+        result = SigningResult(
+            input_digest=input_digest.hex(),
+            cert_pem=b64_cert.decode(),
+            b64_signature=b64_artifact_signature,
+            log_entry=log_entry,
+        )
+
+        signing_result["signature"] = result.b64_signature
+        signing_result["certificate"] = result.cert_pem
+        if self.sigstore_bundle and result.bundle:
+            signing_result["bundle"] = result._to_bundle().to_json()
+
+    def sigstore_verify(self, sha256sumfile, sha256sumsig, sha256sumcert, sigstore_bundle, entry):
+        """Verify a Sigstore signature validity."""
+        if self.verify_offline and not sigstore_bundle:
+            raise ValueError("Offline verification requires a Sigstore bundle.")
+
+        if self.verify_offline and sigstore_bundle:
+            verification_materials = VerificationMaterials.from_bundle(
+                input_=sha256sumfile, bundle=sigstore_bundle, offline=True
+            )
+        else:
+            verification_materials = VerificationMaterials(
+                input_=sha256sumfile,
+                cert_pem=sha256sumcert,
+                signature=sha256sumsig,
+                rekor_entry=entry,
+                offline=False,
+            )
+
+        policy = Identity(
+            identity=self.cert_identity,
+            issuer=self.expected_identity_provider,
+        )
+        return self.verifier.verify(materials=verification_materials, policy=policy)
+
+    class Meta:
+        default_related_name = "%(app_label)s_%(model_name)s"
+
+
+class CollectionVersionSigstoreSignature(Content):
```

Comment:
Is there a useful way to have only one CollectionVerisionSignature Content type?

@rochacbruno I think this involves some questions around how the fact that a collection is signed is represented in galaxy.

Reply:
@mdellweg @mayaCostantini 

Right now galaxy shows a `signed` badge and allows filtering only signed collections. 

To determine that state it only counts the number of signatures in the listed repo for the specified collection.

There are plans to show more information about the signature in the future but none implemented yet